wsd/RequestVettingStation.cpp

```diff
@@ -141,6 +144,71 @@ void RequestVettingStation::sendUnauthorizedErrorAndShutdown()
                          WebSocketHandler::StatusCodes::POLICY_VIOLATION);
 }
 
+#if !MOBILEAPP
+
+namespace
+{
+
+struct SharedSettings
+{
+    SharedSettings(const Poco::JSON::Object::Ptr wopiInfo)
+    {
+        if (auto settingsJSON = wopiInfo->getObject("SharedSettings"))
+        {
+            JsonUtil::findJSONValue(settingsJSON, "uri", _uri);
+            _configId = Cache::getConfigId(_uri);
+
+            std::string stamp;
+            JsonUtil::findJSONValue(settingsJSON, "stamp", stamp);
+            if (!stamp.empty())
+                _configId.append("-").append(stamp);
+        }
+    }
+
+    std::string _uri;
+    std::string _configId;
+
+    std::string getConfigId() const
+    {
+        return _configId;
+    }
+};
+
+}
+
+void RequestVettingStation::launchInstallPresets()
+{
+    SharedSettings sharedSettings(_checkFileInfo->wopiInfo());
+    if (sharedSettings._uri.empty())
+        return;
+
+    std::string configId = sharedSettings.getConfigId();
+
+    auto finishedCallback = [selfWeak = weak_from_this(), this, configId](bool success)
+    {
+        std::shared_ptr<RequestVettingStation> selfLifecycle = selfWeak.lock();
+        if (!selfLifecycle)
+            return;
+
+        if (!success)
+        {
+            sendErrorAndShutdown(_ws, "shared config install failed",
+                                 WebSocketHandler::StatusCodes::UNEXPECTED_CONDITION);
+        }
+        else
+        {
+            COOLWSD::ensureSubForKit(configId);
```

Comment:
The implication of the comment above the `DocumentBroker::asyncInstallPresets()` call (line 206) is that here we "launch docbroker," but it's unclear to me how this is related.
Perhaps I need to understand the workflow better.

I suspect these continuations here, in the Vetting Station, now have a large number of combinations of conditions that need testing. I'd mark this file for better review scrutiny and testing.